src/masternode/masternode-sync.cpp

```diff
@@ -211,6 +212,27 @@ void CMasternodeSync::ProcessTick(CConnman& connman)
                 }
             }
 
+            if (nCurrentAsset == MASTERNODE_SYNC_MEMPOOL) {
+                bool fSshouldSync = pnode->nVersion >= 70216 &&
+                                    !pnode->fInbound &&
+                                    gArgs.GetBoolArg("-syncmempool", DEFAULT_SYNC_MEMPOOL) &&
+                                    !netfulfilledman.HasFulfilledRequest(pnode->addr, "mempool-sync");
+
+                if (fSshouldSync) {
+                    netfulfilledman.AddFulfilledRequest(pnode->addr, "mempool-sync");
+                    connman.PushMessage(pnode, msgMaker.Make(NetMsgType::MEMPOOL));
+                    LogPrintf("CMasternodeSync::ProcessTick -- nTick %d nCurrentAsset %d -- syncing mempool from peer=%d\n", nTick, nCurrentAsset, pnode->GetId());
+                }
+                nTriedPeerCount++;
+                if (nTriedPeerCount < vNodesCopy.size()) {
```

Comment:
I could miss something now its late.. if so let me know what, but couldn't this lead to somewhat like a stucking sync process because of https://github.com/dashpay/dash/blob/develop/src/masternode/masternode-sync.cpp#L149?

Reply:
Good catch!